File: csharp/resistor-color-duo/ResistorColorDuo.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColorDuo
{
    private enum Color
    {
        black, brown, red, orange, yellow, green, blue, violet, grey, white,
    }

    private static readonly Dictionary<string, int> _colorValues;

    static ResistorColorDuo() 
        => _colorValues = ((Color[])Enum.GetValues(typeof(Color)))
        .ToDictionary(color => color.ToString(), color => (int)color);

    public static int Value(string[] colors)
        => 10 * _colorValues[colors[0]] + _colorValues[colors[1]];

}

File: python/resistor-color-duo/resistor_color_duo.py
import enum


class ResistorColor(enum.Enum):
    BLACK = 0
    BROWN = 1
    RED = 2
    ORANGE = 3
    YELLOW = 4
    GREEN = 5
    BLUE = 6
    VIOLET = 7
    GREY = 8
    WHITE = 9


def value(colors: list[str]) -> int:
    first_band, second_band, *_ = colors
    return 10 * color_code(color=first_band) + color_code(color=second_band)


def color_code(color: str) -> int:
    return ResistorColor[color.upper()].value
